### PR TITLE
AddQuote Command

### PR DESCRIPTION
**Pull request by zaydtahir**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #55 

## What is the new behavior?

- Added the add quote command
-
-

### CODE REVIEW

File: src/commands/Quotes/addQuote.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    const response = await message.reply({
+      embeds: [embed],
+      components: [row],
+    });
+  }
+
+  async onLoad() {
+    client.on('interactionCreate', async interaction => {
+      if (!interaction.isButton() || !(interaction.customId === 'QuoteApprove')) return;
+      await interaction.deferUpdate();
```
Comment (sahil-kale) [inline src/commands/Quotes/addQuote.ts:101]: Comment on this function please

File: src/commands/Quotes/addQuote.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+      const message = interaction.message;
+      if (!(message instanceof Message)) return;
+
+      let description = message.embeds[0].description;
+      if (description?.includes(interaction.user.toString())) return;
+
+      const title = message.embeds[0].title;
+      let numApprovals = parseInt(title?.substring('Add Quote Reply - Approvals: '.length, title.indexOf('/')) as string, 10);
```
Comment (sahil-kale) [inline src/commands/Quotes/addQuote.ts:112]: The 'Add Quote Reply - Approvals:` should be coded in somewhere so that if we change one part later it won't look weird and we won't have to update it twice

File: src/commands/Quotes/addQuote.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        const quote = description?.substring('Quote: `'.length, description?.indexOf('`', 'Quote: `'.length + 1));
+        const authorIndex = (description?.indexOf('Author: `') as number) + 'Author: `'.length;
+        const author = description?.substring(authorIndex, description?.indexOf('`', authorIndex + 1));
+
+        await addQuote(interaction.guildId as string, quote as string, author as string);
+
+        await message.edit({
+          embeds: [embed],
+          components: [],
+        });
```
Comment (sahil-kale) [inline src/commands/Quotes/addQuote.ts]: What prevents the else statement from executing if the number of approvals goes over 4?

File: src/commands/Quotes/addQuote.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      }
+    });
+  }
+};
+
+const invalidSyntaxReply = async (message: Message, settingsData: { prefix: string; }) => {
+  logger.info('this happened');
```
Comment (sahil-kale) [inline src/commands/Quotes/addQuote.ts:159]: What does this log statement serve in production code?

File: src/commands/Verification/adminVerify.ts [hunk 5]
```diff
@@ -35,14 +35,14 @@ module.exports = class AdminVerifyCommand extends Command {
...
 
     await addRoleToUser(user.value.id, guild, settingsData.verifiedRole);
     await addVerifiedUser(user.value.id, guildId as string, 'Admin Verified');
     const embed = new BediEmbed()
         .setTitle('Admin Verify Reply')
         .setDescription(`${user.value} has been verified.`);
-    return message.channel.send({embeds: [embed]});
+    return message.reply({embeds: [embed]});
   }
 };
```
Comment (sahil-kale) [inline src/commands/Verification/adminVerify.ts:48]: This should be a separate PR

File: src/commands/Verification/confirm.ts [hunk 6]
```diff
@@ -57,15 +57,15 @@ module.exports = class ConfirmCommand extends Command {
...
+      return message.reply({embeds: [embed]});
     }
 
     if (!(await validUniqueKey(author.id, guildId as string, uniqueKey.value))) {
       const embed = new BediEmbed()
           .setColor(colors.ERROR)
           .setTitle('Confirm Reply')
           .setDescription('Invalid key!');
-      return message.channel.send({embeds: [embed]});
+      return message.reply({embeds: [embed]});
```
Comment (sahil-kale) [inline src/commands/Verification/confirm.ts:68]: As above

File: src/database/models/QuoteModel.ts [hunk 7]
```diff
@@ -16,4 +16,13 @@ export const Quote = new Schema({
 
 const quoteModel = model<QuoteI>('Quote', Quote, 'Quotes');
 
+export const addQuote = async (guildId: string, quote: string, author: string) => {
+  await quoteModel.create({
+    guildId: guildId,
+    quote: quote,
+    author: author,
+    date: Date.now(),
```
Comment (sahil-kale) [inline src/database/models/QuoteModel.ts:24]: This would mean that a quote is generated on model generation rather than when the quote is actually quoted, i.e. if someone goes back and approves a quote days old then the timestamp will be in correct

**Review by sahil-kale — CHANGES_REQUESTED** [review]
Pretty good overall. Please remove the minor changes from the addquote PR and do those separately so the commit history is clean

File: src/commands/Quotes/addQuote.ts [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+      const embed = new BediEmbed()
+          .setColor(colors.ERROR)
+          .setTitle('Add Quote Reply')
+          .setDescription(`You are not verified on this server! Run ${surroundStringWithBackTick(settingsData.prefix + 'verify <emailAddress>')}`);
+      return message.reply({embeds: [embed]});
+    }
+    let quote: string | Result<string, UserError>;
+    let quoteAuthor: Result<string, UserError>;
+
+    if (message.reference) {
```
[the remark this answers is not in the document]
Comment (zaydtahir) [reply src/commands/Quotes/addQuote.ts:46]: The message that was replied to

File: src/commands/Quotes/addQuote.ts [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+import colors from '../../utils/colorUtil';
+import {userVerifiedInGuild} from '../../database/models/VerifiedUserModel';
+import {surroundStringWithBackTick} from '../../utils/discordUtil';
+import logger from '../../utils/loggerUtil';
+import {addQuote} from '../../database/models/QuoteModel';
+import {client} from '../../index';
+
+const {Command} = require('@sapphire/framework');
+
+const EMBED_FIELD_MAX_CHAR_LENGTH = 1025;
```
[the remark this answers is not in the document]
Comment (zaydtahir) [reply src/commands/Quotes/addQuote.ts:13]: wait why is it 1025 i think that's an accident lol

File: src/commands/Quotes/addQuote.ts (hunk 3 above)
Comment (zaydtahir) [reply src/commands/Quotes/addQuote.ts]: The else branch removes the button from the message entirely.

**Review by sahil-kale — APPROVED** [review]
